chainer/variable.py

```diff
@@ -161,6 +161,12 @@ def creator(self):
         """Function node that created this variable node."""
         return self._creator
 
+    @creator.setter
+    def creator(self, func):
+        self._creator = func
+        if func is not None:
+            self._rank = func.rank + 1
+
```

Comment:
No, because `_rank` is used for sorting nodes in backward, and the leaf node (including a variable that does not have a creator) is not involved in the sorting.